Sources/Core/SourceFile.swift

```diff
@@ -26,7 +26,29 @@ public struct SourceFile {
 
   /// Creates an instance representing the file at `filePath`.
   public init(contentsOf filePath: URL) throws {
-    let storage = try Storage(filePath) { try String(contentsOf: filePath) }
+    let storage = try Storage(filePath) { try String(contentsOf: filePath)[...] }
+    self.storage = storage
+  }
+
+  /// Creates an instance for the `text` given by a multiline string literal in the given
+  /// `swiftFile`, the literal's textual content (the line after the opening quotes) being
+  /// startLine.
+  ///
+  /// N.B. The text of the instance will literally be what's in the Swift file, including its
+  /// indentation and any embedded special characters, even if the literal itself is not a raw
+  /// literal or has had indentation stripped by the Swift compiler.
+  fileprivate init(diagnosableLiteral text: String, swiftFile: String, startLine: Int)
+    throws
+  {
```

Comment:
```suggestion
  fileprivate init(
    diagnosableLiteral text: String, swiftFile: String, startLine: Int
  ) throws {
```

You've used this style below, which I prefer.

Reply:
I let swift-format do what it wants, and I don't think we should be asking for this kind of formatting change in code reviews, but I'll accept your suggestion.